std/algorithm/mutation.d

```diff
@@ -562,6 +562,17 @@ void fill(Range, Value)(Range range, Value value)
     }
 }
 
+import std.range : ElementEncodingType;
+import std.traits : isSomeChar, isMutable;
+
+/// ditto
+void fill(A, Value)(auto ref A array, auto ref Value value)
+if (isArray!A && isSomeChar!(ElementEncodingType!A) &&
```

Comment:
To be correct, `A` can be any type that supports the `opIndexAssign`or the old `opSlice`. For example  ndslice supports this as well..